tariff/tariff.go

```diff
@@ -87,19 +93,37 @@ func NewConfigurableFromConfig(ctx context.Context, other map[string]interface{}
 func (t *Tariff) run(forecastG func() (string, error), done chan error, interval time.Duration) {
 	var once sync.Once
 
+	// Try to load from cache on startup for solar forecasts
+	if t.cache != nil {
+		if cached, ok := t.cache.Get(interval); ok {
+			t.log.DEBUG.Printf("loaded %d rates from cache", len(cached))
+			// Apply totalPrice to cached rates if needed
+			for i, r := range cached {
+				cached[i] = api.Rate{
+					Value: t.totalPrice(r.Value, r.Start),
+					Start: r.Start.Local(),
+					End:   r.End.Local(),
+				}
+			}
+			mergeRatesAfter(t.data, cached, beginningOfDay())
```

Comment:
How do we know the cache returned enough data?

Reply:
I've added a check that ensures the cached data covers at least the current day